cargo-near/src/abi.rs

```diff
@@ -53,15 +55,34 @@ pub(crate) fn write_to_file(
     if !generate_docs {
         strip_docs(&mut contract_abi);
     }
-    let near_abi_json = serde_json::to_string(&contract_abi)?;
-    let out_path_abi = crate_metadata
-        .target_directory
-        .join(crate_metadata.root_package.name.replace('-', "_") + "_abi.json");
-    fs::write(&out_path_abi, near_abi_json)?;
+    let near_abi_serialized = match format {
+        AbiFormat::Json => serde_json::to_vec_pretty(&contract_abi)?,
+        AbiFormat::JsonMin => serde_json::to_vec(&contract_abi)?,
+    };
+    let near_abi_compressed = match compression {
+        AbiCompression::NoOp => near_abi_serialized,
+        AbiCompression::Zstd => zstd::encode_all(near_abi_serialized.as_slice(), 0)?,
+    };
+
+    let out_path_abi = crate_metadata.target_directory.join(format!(
+        "{}_abi.{}",
+        crate_metadata.root_package.name.replace('-', "_"),
+        abi_file_extension(format, compression)
+    ));
+    fs::write(&out_path_abi, near_abi_compressed)?;
 
     Ok(AbiResult { path: out_path_abi })
 }
 
+fn abi_file_extension(format: AbiFormat, compression: AbiCompression) -> &'static str {
+    match compression {
+        AbiCompression::NoOp => match format {
+            AbiFormat::Json | AbiFormat::JsonMin => "json",
+        },
+        AbiCompression::Zstd => "bin",
```

Comment:
Any reason for not using `.zst`?

Reply:
Nothing really, just wasn't aware that there is a special extension.

Reply:
```console
$ unzst res/abi_abi.bin
zstd: res/abi_abi.bin: unknown suffix (.zst/.tzst/.gz/.tgz/.lzma/.xz/.txz/.lz4/.tlz4 expected).
Can't derive the output file name. Specify it with -o dstFileName. Ignoring.
```

Reply:
Already changed it to `.zst`, but that is a really good point. Haven't tried `unzst` manually.